### ISSUE TITLE
Batch GPT is not working

### ISSUE THREAD

**Issue by NimrodShabtay** [EvolvingLMMs-Lab/lmms-eval#279]
Hi,
Thanks in advance for all the hard work and this amazing project.
I am trying to run `batch_gpt` for a dataset of mine (works on HF models)
But I am getting errors:
1. First it seems the returned batches are binary and not text so you need to extract them before converting to JSON:
```
batch_results = self.retrieve_batch_results(batch_status.output_file_id).text.split('\n')[:-1]
res = [json.loads(result)["response"]["body"]["choices"][0]["message"]["content"] for result in batch_results]
```
After I managed to get the data I got an error:
```
lmms-eval/lmms_eval/filters/selection.py", line 16, in <lambda>
    return map(lambda r: r[0], resps)
    IndexError: list index out of range
```

so maybe my solution was not good enough

Any suggestions?

**Comment by Luodian** [comment]
Yes, actually this is a trial and wish to finish later, it's just not working.

**Comment by abzb1** [comment]
You’re making a very good point. It seems that what retrieve_batch_results() is returning is an object, so it’s not immediately usable in the desired data format. It looks like some adjustments need to be made. Separately, in my experience, some requests in a batch GPT request can fail. In such cases, the size of the benchmark dataset and the length of the returned response may not match.
